### PR TITLE
Change SVT VP9 external API names to avoid conflicts with AV1

### PR DESCRIPTION
**Pull request by jsunintel**

### CODE REVIEW

**Comment by jsunintel** [comment]
@tianjunwork @hassount

**Comment by hassount** [comment]
@jsunintel 
I don't follow, so this patch removes the EbApi.h file but it does not replace it with EbSvtVp9Api.h ? 
also, can we follow the SVT-AV1 nomenclature and carry it over to SVT-HEVC and SVT-VP9?

**Comment by jsunintel** [comment]
> 
> 
> @jsunintel
> I don't follow, so this patch removes the EbApi.h file but it does not replace it with EbSvtVp9Api.h ?
> also, can we follow the SVT-AV1 nomenclature and carry it over to SVT-HEVC and SVT-VP9?

This patch renamed EbXXX.h to EbSvtVp9XXX.h, following how AV1 had enamed its API header file names.

**Comment by hassount** [comment]
gotcha, I just don't see the actual files being renamed, I just see EbApi.h being removed (this is why the ffmpeg patch is failing I believe). and for the naming, can we do EbSvtVp9Enc.h instead of EbSvtVp9Api.h

**Comment by jsunintel** [comment]
> 
> 
> gotcha, I just don't see the actual files being renamed, I just see EbApi.h being removed (this is why the ffmpeg patch is failing I believe along with the windows build).
> For the naming, can we do EbSvtVp9Enc.h instead of EbSvtVp9Api.h

That is kind of confusing, but Git just works in that way and we can't change it.
OK, I am submitting that EbSvtVp9Enc.h->EbSvtVp9Api.h change.

**Comment by jsunintel** [comment]
Hi @hassount please review the updated again and merge it if no more issue.

File: Source/API/EbSvtVp9Enc.h [hunk 1]
```diff
@@ -353,7 +353,7 @@ typedef struct EbSvtVp9EncConfiguration
 * @ *p_app_data      Callback data.
 * @ *config_ptr     Pointer passed back to the client during callbacks, it will be
 *                  loaded with default params from the library. */
-EB_API EbErrorType eb_svt_init_handle(
+EB_API EbErrorType eb_vp9_svt_init_handle(
     EbComponentType** p_handle,
```
Comment (tianjunwork) [inline Source/API/EbSvtVp9Enc.h]: how about eb_svt_vp9_xxx?

File: gstreamer-plugin/gstsvtvp9enc.c [hunk 2]
```diff
@@ -329,9 +329,9 @@ gst_svtvp9enc_init (GstSvtVp9Enc * svtvp9enc)
   svtvp9enc->dts_offset = 0;
 
   EbErrorType res =
-      eb_svt_init_handle(&svtvp9enc->svt_encoder, NULL, svtvp9enc->svt_config);
+      eb_vp9_svt_init_handle(&svtvp9enc->svt_encoder, NULL, svtvp9enc->svt_config);
```
Comment (tianjunwork) [inline gstreamer-plugin/gstsvtvp9enc.c]: Thanks Jing also updated gst plugin:)

**Comment by tianjunwork** [comment]
Hi @jsunintel, could fix Travis error?

File: Source/API/EbSvtVp9Enc.h (hunk 1 above)
Comment (jsunintel) [reply Source/API/EbSvtVp9Enc.h]: > 
> 
> how about eb_svt_vp9_xxx? or simply eb_vp9_xxx?
> @hassount what do you think?

I think those are basically the same:-)
@hassount would you please review this?

**Comment by jsunintel** [comment]
> 
> 
> Hi @jsunintel, could fix Travis error?

Those are not real errors. As I mentioned above, caused by API changes.

**Comment by 1480c1** [comment]
Can you make sure the ffmpeg patches work by using git am *.patch? When I tried, it said that a lot of the changes did not apply/ weren't applied. Next is there an order to the patches? Also I see that one has SvtAv1 in it even though the file name doesn't mention it

**Comment by 1480c1** [comment]
Also, are the patches generated from ffmpeg head or 4.1?

**Comment by jsunintel** [comment]
> 
> 
> Can you make sure the ffmpeg patches work by using git am *.patch? When I tried, it said that a lot of the changes did not apply/ weren't applied. Next is there an order to the patches? Also I see that one has SvtAv1 in it even though the file name doesn't mention it

Updated the readme:
https://github.com/OpenVisualCloud/SVT-VP9/pull/25/files#diff-75127208d35ef28826fb972fb3b0a310

Both the 2 branches work

**Pull request merged**